docs/source/extending.rst

```diff
@@ -0,0 +1,144 @@
+.. _extending:
+
+======================
+Extending the Notebook
```

Comment:
I think the title should possibly be more specific: 'Extending the Notebook Server'? Nbextensions also extend the notebook, and that's a very different subject.